src/backends/mock_main.rs

```diff
@@ -278,7 +287,7 @@ impl MockMainPod {
             Self::pad_operation_args(params, &mut mid_args);
             let mut args = Vec::with_capacity(mid_args.len());
             for mid_arg in &mid_args {
-                args.push(Self::find_op_arg(statements, mid_arg));
+                args.push(Self::find_op_arg(statements, mid_arg).expect("Failed to find operation argument"));
```

Comment:
since you added at the `find_op_arg` method to return an error (line 248), maybe this current line could return that error to the upper level instead of panicking by the `.expect`?

(same for the `expect` at line 312)